ProgramChainid uses hardcoded chain_id instead of chain_id passed in through commandline arguments

This test is failing for me in `execute remote` mode when I am trying to test against Arbitrum using `tests/frontier/scenarios/test_scenarios.py::test_scenarios[fork_Osaka-blockchain_test-test_program_program_CHAINID-debug]`

The test gives me result `- AssertionError: Storage value at 0x01 of 0x89253c2b35afc86022e253eaa224d607669dc7dc is 0x0000000000000000000000000000000000000000000000000000000000064aba,expected 0x01`

This seems due to 

https://github.com/ethereum/execution-spec-tests/blob/508fa1e231539476a3fba98fd4ee418b31e04324/tests/frontier/scenarios/programs/context_calls.py#L416-L430

<img width="957" height="603" alt="Image" src="https://github.com/user-attachments/assets/b9123999-40d6-4d53-a60b-e72ed0768e65" />

Hardcoding a chain_id of 1, we should update this program to use https://github.com/ethereum/execution-spec-tests/pull/2084 the chain_id configured by ``--rpc-chain-id``

--- a/tests/frontier/scenarios/programs/context_calls.py
+++ b/tests/frontier/scenarios/programs/context_calls.py
@@ -430,8 +430,7 @@ class ProgramChainid(ScenarioTestProgram):
 
     def result(self) -> ProgramResult:
         """Test result."""
-        # TODO: why attribute error when accessing ChainConfig.chain_id?
-        #       only ChainConfigDefaults works
+        # TODO: use `chain_config` fixture instead.
         chain_id = ChainConfigDefaults.chain_id
 
         return ProgramResult(result=chain_id, from_fork=Istanbul)

